python-avd/pyavd/_utils/format_as_path_list_regex.py

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+def format_as_path_list_regex(regex: str) -> str:
+    """
+    Format the given regex for an as-path access-list to conform to the regex rules.
+
+    For now this only escapes dots in case of dotted BGP ASN.
+    """
+    return regex.replace(".", "\\.")
```

Comment:
I am not sure this is quite right to directly pass a regex here. `.` does mean something in the regex style (any ASN) as per documentation and so what we should escape in the regex are only the asdot formatted ASN

Reply:
something like `"\d+\.\d+"?`